search/searcher/search_synonym.go

```diff
@@ -45,7 +45,7 @@ func tokenizeSynonym(phrase []byte) []string {
 	return rv
 }
 
-type ctxKey string
+type synonymSearchCtxKey string
```

Comment:
what i meant in the earlier comment was to have something like this:

`const synonymSearchCtxKey = "_searcher_positions"`

and have `context.WithValue(ctx, synonymSearchCtxKey, searchersPositions)` down below, the reason why i'm suggesting this particular way is because in the code that's how the context keys are defined, so its better to not break that pattern i guess.